### PR TITLE
Fix rocksdb lock panic and add container chain monitor

### PR DESCRIPTION
**Pull request by tmpolaczyk**
Fixes rocksdb lock panic by using a patched version of jsonrpsee.

Adds a "container chain monitor" that will warn when a node is running more container chains than expected, and can be used to debug the cause of why the container chains haven't stopped properly.

### CODE REVIEW

File: node/src/service.rs [hunk 1]
```diff
@@ -169,6 +169,14 @@ pub fn new_partial(
...
         )?;
     let client = Arc::new(client);
+    log::info!(
+        "backend in new_partial refcount: {}",
+        Arc::strong_count(&backend)
+    );
+    log::info!(
+        "client in new_partial refcount: {}",
+        Arc::strong_count(&client)
+    );
```
Comment (tmpolaczyk) [inline node/src/service.rs:179]:
```suggestion
```

File: node/src/service.rs [hunk 2]
```diff
@@ -626,6 +634,46 @@ async fn start_node_impl(
             }
         }
 
+        let debug_start_same_cc = async move {
```
Comment (tmpolaczyk) [inline node/src/service.rs:637]: Also remove this

File: node/src/service.rs [hunk 3]
```diff
@@ -709,6 +765,14 @@ pub async fn start_node_impl_container(
...
         params_import_queue = params.import_queue;
     }
+    log::info!(
+        "backend start_node_impl_container refcount: {}",
+        Arc::strong_count(&backend)
+    );
+    log::info!(
+        "client start_node_impl_container refcount: {}",
+        Arc::strong_count(&client)
+    );
```
Comment (tmpolaczyk) [inline node/src/service.rs:775]:
```suggestion
```

**Comment by girazoki** [comment]
@tmpolaczyk I took the chance of pointing at the right repo and fixing clippy, please re-visist that I did not mess up

**Comment by tmpolaczyk** [comment]
>@tmpolaczyk I took the chance of pointing at the right repo and fixing clippy, please re-visist that I did not mess up

Looks good, thanks! The reason I decided to use a patch section in Cargo.toml instead of modifying the repo url for all of our dependencies, like we usually do when we fork substrate, is because everything depends on jsonrpsee, so we would have needed to change all the repos.

**Comment by girazoki** [comment]
yeah I think its the right approach!

File: node/src/service.rs [hunk 4]
```diff
@@ -164,6 +164,7 @@ pub fn new_partial(
         .build();
 
     let executor = ParachainExecutor::new_with_wasm_executor(wasm);
+    //let wasm_cache = executor.cache;
```
Comment (girazoki) [inline node/src/service.rs:167]:
```suggestion
```

**Review by girazoki — APPROVED** [review]

**Pull request merged**